dpgen/data/gen.py

```diff
@@ -237,12 +237,24 @@
 def poscar_scale(poscar_in, poscar_out, scale):
     with open(poscar_in) as fin:
         lines = list(fin)
-    if "D" == lines[7][0] or "d" == lines[7][0]:
+
+    # Determine if "Selective dynamics" is present
+    if lines[7].strip().lower().startswith("s"):
+        coord_type_line = 8  # If present, coordinates type is on line 9
+    else:
+        coord_type_line = 7  # If not, coordinates type is on line 8
+
+    # Process according to the coordinate type
+    if "D" == lines[coord_type_line][0] or "d" == lines[coord_type_line][0]:
         lines = poscar_scale_direct(lines, scale)
-    elif "C" == lines[7][0] or "c" == lines[7][0]:
+    elif "C" == lines[coord_type_line][0] or "c" == lines[coord_type_line][0]:
         lines = poscar_scale_cartesian(lines, scale)
```

Comment:
**Suggestion:** Use the `str.startswith()` method to check the starting character of a string in a case-insensitive manner, instead of checking equality with both the uppercase and lowercase versions of the character. [best practice]
```suggestion
    if lines[coord_type_line].startswith(("D", "d")):
        lines = poscar_scale_direct(lines, scale)
    elif lines[coord_type_line].startswith(("C", "c")):
        lines = poscar_scale_cartesian(lines, scale)
```